lib/email/receiver.rb

```diff
@@ -444,6 +446,17 @@ def trim_discourse_markers(reply)
     end
 
     def parse_from_field(mail)
+      if mail.bounced? || verp
+        final_recipient = mail.final_recipient
+        return extract_from_address_and_name(final_recipient) if final_recipient.is_a? String
```

Comment:
I like to use `Array.wrap`, because you don't need to distinguish between `String` and `Array`.